lib/SensorFusion/src/SensorFusion.cpp

```diff
@@ -58,35 +58,82 @@
  * Public Methods
  *****************************************************************************/
 
-void SensorFusion::init(void)
-{
-    m_linearKalmanFilter.init();
-}
-
 void SensorFusion::estimateNewState(const SensorData& newSensorData)
 {
     /* Calculate the physical Values via the Sensitivity Factors. */
-    int16_t physicalAccelerationX = newSensorData.accelerationX * SensorConstants::ACCELEROMETER_SENSITIVITY_FACTOR;
-    int16_t physicalAccelerationY = newSensorData.accelerationY * SensorConstants::ACCELEROMETER_SENSITIVITY_FACTOR;
-    int16_t physicalTurnRate      = newSensorData.turnRate * SensorConstants::GYRO_SENSITIVITY_FACTOR;
-
-    /* Transform the acceleration values from the robot coordinate system into the world coordinate system */
-    int16_t accelerationInRobotCoordinateSystem[2] = {physicalAccelerationX, physicalAccelerationY};
-    int16_t accelerationInGlobalCoordinateSystem[2];
-    transformLocalToGlobal(accelerationInGlobalCoordinateSystem, accelerationInRobotCoordinateSystem,
-                           m_estimatedPosition.angle);
+    float physicalAccelerationX = static_cast<float>(newSensorData.accelerationX) *
+                                  SensorConstants::ACCELEROMETER_SENSITIVITY_FACTOR; /* In mm/s^2 */
+    float physicalTurnRate =
+        static_cast<float>(newSensorData.turnRate) * SensorConstants::GYRO_SENSITIVITY_FACTOR; /* In mrad/s */
 
-    /* Perform the Kalman Filter Prediction and Update Steps */
     KalmanParameter kalmanParameter;
-    kalmanParameter.accelerationX     = physicalAccelerationX;
-    kalmanParameter.accelerationY     = physicalAccelerationY;
-    kalmanParameter.positionOdometryX = newSensorData.positionOdometryX;
-    kalmanParameter.positionOdometryY = newSensorData.positionOdometryY;
-    kalmanParameter.angleOdometry     = newSensorData.orientationOdometry;
-    kalmanParameter.turnRate          = physicalTurnRate;
-
-    m_linearKalmanFilter.predictionStep(newSensorData.timePeriod);
-    m_estimatedPosition = m_linearKalmanFilter.updateStep(kalmanParameter);
+    if (true == m_isFirstIteration)
+    {
+        /* Directely use the Position calculated via Odometry in the first iteration */
+        kalmanParameter.positionOdometryX = static_cast<float>(newSensorData.positionOdometryX);   /* In mm */
+        kalmanParameter.positionOdometryY = static_cast<float>(newSensorData.positionOdometryY);   /* In mm */
+        kalmanParameter.angleOdometry     = static_cast<float>(newSensorData.orientationOdometry); /* In mrad */
+        kalmanParameter.turnRate          = physicalTurnRate;                                      /* In mrad/s */

```

Comment:
kalmanParameter.turnRate & kalmanParameter.accelerationX seem not to be assigned different values in the if-branches. Therefore it could be assigned outside (i.e. before) the if.